Sources/Testing/Parameterization/Test.Case.swift

```diff
@@ -55,13 +78,16 @@ extension Test {
     ///   - values: The values passed to the parameters for this test case.
     ///   - parameters: The parameters of the test function for this test case.
     ///   - body: The body closure of this test case.
+    ///
+    /// - Throws: Any error encountered attempting to encode test arguments.
     init(
       values: [any Sendable],
       parameters: [ParameterInfo],
       body: @escaping @Sendable () async throws -> Void
-    ) {
-      let arguments = zip(values, parameters).map { value, parameter in
-        Argument(value: value, parameter: parameter)
+    ) throws {
+      let arguments = try zip(values, parameters).map { value, parameter in
```

Comment:
If an error occurs here, it might not be a test failure. For instance, if a particular input is of a codable type but is not safely encodable in this context (e.g. XPC-codable-only types?), we shouldn't fail the test, we should just fail to identify the argument (thus it's not re-runnable independently, but it's still _testable_.)

Reply:
Ok I can buy that. I do want to capture and reflect the error back to the user somehow, so they know that there was an error during encoding that will prevent them from re-running, but we can track that as a separate enhancement.